test/inductor/test_cpu_repro.py

```diff
@@ -2466,6 +2466,28 @@ def fn(x, y):
         y = torch.randn(8, 8, 3136, 8)
         self.common(fn, (x, y))
 
+    @unittest.skipIf(not torch.backends.mkldnn.is_available(), "MKLDNN is not enabled")
+    @patch("torch.cuda.is_available", lambda: False)
+    @config.patch(freezing=True)
+    def test_linear_with_no_default_contiguous_input(self):
+        mod = torch.nn.Sequential(torch.nn.Linear(16, 16)).eval()
+        temp = torch.randn(1, 16, 1, 1)
+        v = torch.ops.inductor._reinterpret_tensor(temp, [1, 16], [0, 1], 0)
```

Comment:
Nit: why using `_reinterpret_tensor` instead of `as_strided`? `_reinterpret_tensor` is meant to be an internal op.

Reply:
Updated, using as_strided now.